tokio/src/macros/join.rs

```diff
@@ -0,0 +1,109 @@
+/// Wait on multiple concurrent branches, returning when **all** branches
```

Comment:
Do you think it's worth mentioning the `futures` crate's `join` and noting that the macro is similar but allows joining a variadic set of futures without requiring a `Vec`?

Reply:
(e.g. it would be nice to note that the variadic nature of the macro is handled at compile time by generating fixed size tuples, rather than allocating a vec...)

Reply:
The futures` version does not require a vec, but adding a note is :+1:

Reply:
> The futures` version does not require a vec,
`join_all` doesn't allocate a vec?

Reply:
`join_all` does but there is also a `join!` macro